plugins/inputs/logfile/logfile.go

```diff
@@ -398,22 +398,17 @@ func (t *LogFile) cleanUpStoppedTailerSrc() {
 	}
 }
 
+// checkForDuplicateRetentionSettings: Checks if a log group has retention set in multiple places and stops the agent if found
 func (t *LogFile) checkForDuplicateRetentionSettings() {
 	configMap := make(map[string]int)
-
 	for i := range t.FileConfig {
-		fileconfig := &t.FileConfig[i]
-		if fileconfig.LogGroupName != "" {
-			logGroup := strings.ToLower(fileconfig.LogGroupName)
+		fileConfig := &t.FileConfig[i]
+		if fileConfig.LogGroupName != "" && fileConfig.RetentionInDays !=-1 {
+			logGroup := strings.ToLower(fileConfig.LogGroupName)
 			configMap[logGroup] += 1
 		}
-
-	}
-	for i := range t.FileConfig {
-		fileconfig := &t.FileConfig[i]
-		// log group has Retention settings in multiple places: throw an error
-		if fileconfig.LogGroupName != "" && configMap[strings.ToLower(fileconfig.LogGroupName)] > 1 {
-			panic(fmt.Sprintf("error: retention for the same log group set in multiple places. Log Group Name: %v", fileconfig.LogGroupName))
+		if fileConfig.LogGroupName != "" && configMap[strings.ToLower(fileConfig.LogGroupName)] > 1 {
```

Comment:
If we intend to panic whenever we have more than one non-default value for the retention period for any given log group, I think we can restructure this check rather than just dropping the two existing if-statements into the loop. I feel like having the count is unnecessary if we're just concerned about the existence of at least two conflicting configurations. 

1. Define a `map[string]bool` for marking LogGroupName -> HasNonDefaultRetention
2. Iterate over each of the file configurations
3. If the retention policy is non-default (>= -1), then check if the log group name is in the map. If it is not, put it in the map, else panic

```go
configMap := make(map[string]bool)
for i := range t.FileConfig {
    fileConfig := &t.FileConfig[i]
    if fileConfig.RetentionInDays != -1 {
        if _, ok := configMap[fileConfig.LogGroupName]; !ok {
            configMap[fileConfig.LogGroupName] = true
        } else {
             panic()
        }
    }
}
```
or something like that. Obviously, if we factor in the other review comment I had about two equivalent non-default retention periods in the configuration, this would change.